apollo-router/src/configuration/mod.rs

```diff
@@ -246,37 +295,186 @@ impl JsonSchema for UserPlugins {
+            path: path.unwrap_or_else(default_graphql_path),
+            introspection: introspection.unwrap_or_else(default_introspection),
+            preview_defer_support: preview_defer_support.unwrap_or_else(default_defer_support),
+        }
+    }
+}
+
+impl Default for Graphql {
+    fn default() -> Self {
+        Self::builder().build()
+    }
+}
+
+/// Configuration options pertaining to the http server component.
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct Sandbox {
+    /// The socket address and port to listen on
+    /// Defaults to 127.0.0.1:4000
+    #[serde(default = "default_graphql_listen")]
+    pub(crate) listen: ListenAddr,
 
     /// The HTTP path on which GraphQL requests will be served.
     /// default: "/"
     #[serde(default = "default_graphql_path")]
-    pub(crate) graphql_path: String,
+    pub(crate) path: String,
+
+    #[serde(default = "default_sandbox")]
+    pub(crate) enabled: bool,
+}
 
-    /// healthCheck path
-    /// default: "/.well-known/apollo/server-health"
+fn default_sandbox() -> bool {
+    true
+}
+
+#[buildstructor::buildstructor]
+impl Sandbox {
+    #[builder]
+    pub(crate) fn new(
+        listen: Option<ListenAddr>,
+        path: Option<String>,
+        enabled: Option<bool>,
+    ) -> Self {
+        Self {
+            listen: listen.unwrap_or_else(default_graphql_listen),
+            path: path.unwrap_or_else(default_graphql_path),
+            enabled: enabled.unwrap_or_else(default_sandbox),
+        }
+    }
+
+    // Used in tests
+    #[allow(dead_code)]
+    #[builder]
+    pub(crate) fn fake_new(
+        listen: Option<ListenAddr>,
+        path: Option<String>,
+        enabled: Option<bool>,
+    ) -> Self {
+        Self {
+            listen: listen.unwrap_or_else(test_listen),
+            path: path.unwrap_or_else(default_graphql_path),
+            enabled: enabled.unwrap_or_else(default_sandbox),
+        }
+    }
+}
+
+impl Default for Sandbox {
+    fn default() -> Self {
+        Self::builder().build()
+    }
+}
+
+/// Configuration options pertaining to the http server component.
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct HealthCheck {
+    /// The socket address and port to listen on
+    /// Defaults to 127.0.0.1:9090
+    #[serde(default = "default_health_check_listen")]
+    pub(crate) listen: ListenAddr,
+
+    /// The HTTP path on which GraphQL requests will be served.
+    /// default: "/"
     #[serde(default = "default_health_check_path")]
-    pub(crate) health_check_path: String,
+    pub(crate) path: String,
+}
 
-    /// Preview @defer directive support
-    /// default: true
-    #[serde(default = "default_defer_support")]
-    pub(crate) preview_defer_support: bool,
+fn default_health_check_listen() -> ListenAddr {
+    SocketAddr::from_str("127.0.0.1:9090").unwrap().into()
```

Comment:
wasn't the 9090 port supposed to be used only by the prometheus endpoint?

Reply:
I don't think so, the [comment i ve worked from](https://github.com/apollographql/router/issues/1500#issuecomment-1238216322) says otherwise

Reply:
Originally the health check was on the same port as the graphql server, that's why it was under the `.well-known` path

Reply:
yes, this is an intentional breaking change, the goal is to not expose it to internet by default.